libs/ngx-nuts-and-bolts/src/lib/table-state/table-state.helper.ts

```diff
@@ -0,0 +1,153 @@
+import { ActivatedRoute, Router } from '@angular/router';
+import { isEqual } from 'lodash';
+import { Observable } from 'rxjs';
+import { distinctUntilChanged, map } from 'rxjs/operators';
+
+export type Sort = 'asc' | 'desc' | '';
+
+export interface IPageInfo {
+	pageIndex: number;
+	pageSize: number;
+	length?: number;
+}
+
+export interface ISortInfo {
+	sortingKey: string;
+	sortDirection: Sort;
+}
+
+export enum TableQueryParam {
+	PAGE_INDEX = 'pageIndex',
+	PAGE_SIZE = 'pageSize',
+	SORT_KEY = 'sortKey',
+	SORT_DIRECTION = 'sortDirection',
+	CUSTOM_FILTERS = 'filters',
+}
+
+export function createPaginationObservable(route: ActivatedRoute): Observable<IPageInfo> {
+	return route.queryParamMap.pipe(
+		map((paramMap) => {
+			const pageIndex = parseInt(paramMap.get(TableQueryParam.PAGE_INDEX) ?? '0', 10);
+			const pageSize = parseInt(paramMap.get(TableQueryParam.PAGE_SIZE) ?? '10', 10);
+
+			return {
+				pageIndex,
+				pageSize,
+			};
+		}),
+		distinctUntilChanged(
```

Comment:
I would let consumers do the distinctUntilChanged further down the line if they so desire.

Reply:
I belive this was added so if route changes (maybe some other query params are added) this prevents refetching od the data. It emits only if parameters on which api calls depend are changed. To me it makes sense to be part of this logic, but I could be wrong

Reply:
As mentioned in [this comment](https://github.com/infinum/ngx-nuts-and-bolts/pull/31#discussion_r792500069), we can leave this up to the users and that way we also avoid having to depend on lodash